libs/ui-lib/lib/ocm/components/featureSupportLevels/FeatureSupportLevelProvider.tsx

```diff
@@ -123,80 +70,81 @@ export const FeatureSupportLevelProvider: React.FC<SupportLevelProviderProps> =
     [cluster?.cpuArchitecture, infraEnv?.cpuArchitecture, infraEnv?.staticNetworkConfig],
   );
 
-  const getVersionSupportLevelsMapCallback = React.useCallback(
-    (versionName: string): FeatureIdToSupportLevel | undefined => {
-      return getVersionSupportLevelsMap(versionName, supportLevelData);
-    },
-    [supportLevelData],
-  );
+  const getFeatureSupportLevels = React.useCallback((): NewFeatureSupportLevelMap => {
+    return supportLevelData;
+  }, [supportLevelData]);
 
   const getFeatureSupportLevel = React.useCallback(
-    (versionName: string, featureId: FeatureId): SupportLevel | undefined => {
-      const versionSupportLevelData = getVersionSupportLevelsMapCallback(versionName);
-      return versionSupportLevelData ? versionSupportLevelData[featureId] : undefined;
+    (
+      featureId: FeatureId,
+      supportLevelDataNew?: NewFeatureSupportLevelMap,
+    ): SupportLevel | undefined => {
       // eslint-disable-next-line react-hooks/exhaustive-deps
+      if (supportLevelDataNew) {
```

Comment:
I am not properly understanding this.
Why do we sometimes use some data vs the other?

Reply:
When we are using in NewClusterPage, because data is changing when user use the Openshift version dropdown and CPU architecture dropdown.